tools/ot/src/test/java/dk/alexandra/fresco/tools/ot/otextension/PseudoOtpTest.java

```diff
@@ -0,0 +1,134 @@
+  }
+
+  @Test
+  public void testZeroMessageLength() {
+    int candidateLength = 50;
+    int messageLength = 0;
+    int cipherLength = 100;
+    testEncryptDecrypt(messageLength, candidateLength, cipherLength);
+  }
+
+  @Test
+  public void testZeroCandidateLength() {
+    int candidateLength = 0;
+    int messageLength = 40;
+    int cipherLength = 100;
+    testEncryptDecrypt(messageLength, candidateLength, cipherLength);
+  }
+
+  @Test
+  public void testZeroCipherLength() {
+    int candidateLength = 50;
+    int messageLength = 40;
+    int cipherLength = 0;
+    testEncryptDecrypt(messageLength, candidateLength, cipherLength);
+  }
+
+  @Test (expected = IllegalArgumentException.class)
+  public void testNegativeCipherLength() {
+    int candidateLength = 50;
+    int messageLength = 40;
+    int cipherLength = -100;
+    testEncryptDecrypt(messageLength, candidateLength, cipherLength);
+  }
+
+  @Test (expected = NullPointerException.class)
+  public void testEncryptNullMessage() {
+    PseudoOtp.encrypt(null, new byte[] { 0x00 });
+  }
+
+  @Test (expected = NullPointerException.class)
+  public void testEncryptNullCandidate() {
+    PseudoOtp.encrypt(new byte[] { 0x00 }, null);
+  }
+
+  @Test (expected = NullPointerException.class)
+  public void testDecryptNullCipher() {
+    PseudoOtp.decrypt(null, new byte[] { 0x00 });
+  }
+
+  @Test (expected = NullPointerException.class)
+  public void testDecryptNullCandidate() {
+    PseudoOtp.encrypt(new byte[] { 0x00 }, null);
+  }
+
+  private void testEncryptDecrypt(int messageLength, int candidateLength) {
+    Random rand = new Random();
+    byte[] message = new byte[messageLength];
+    rand.nextBytes(message);
+    byte[] candidate = new byte[candidateLength];
+    rand.nextBytes(candidate);
+    byte[] cipherText = PseudoOtp.encrypt(message, candidate);
+    assertEquals(messageLength, cipherText.length);
+    assertFalse(Arrays.equals(message, cipherText));
+    assertFalse(Arrays.equals(Arrays.copyOf(candidate, messageLength), cipherText));
+    byte[] decryptedMessage = PseudoOtp.decrypt(cipherText, candidate);
+    assertArrayEquals(message, decryptedMessage);
+  }
+
+  private void testEncryptDecrypt(int messageLength, int candidateLength, int cipherLength) {
+    Random rand = new Random();
+    byte[] message = new byte[messageLength];
+    rand.nextBytes(message);
+    byte[] candidate = new byte[candidateLength];
+    rand.nextBytes(candidate);
+    byte[] cipherText = PseudoOtp.encrypt(message, candidate, cipherLength);
+    assertEquals(cipherLength, cipherText.length);
+    if (cipherText.length > 0) {
+     assertFalse(Arrays.equals(Arrays.copyOf(message, cipherLength), cipherText));
+     assertFalse(Arrays.equals(Arrays.copyOf(candidate, cipherLength), cipherText));
+    }
+    byte[] decryptedMessage = PseudoOtp.decrypt(cipherText, candidate, cipherLength);
+    assertArrayEquals(Arrays.copyOf(message, cipherLength), decryptedMessage);
```

Comment:
Perhaps we should also check that no zero-vectors have sneaked in here?

Reply:
I think that is already covered by testing for equality of the original message.

Reply:
I agree, handled with the extra check in the lengthAdjustmentTest